### DISCUSSION TITLE
ESP32-S2 / S3 / C3 -- Uploading sketches using CDC / DFU option

### DISCUSSION THREAD

**Discussion by SuGlider** [espressif/arduino-esp32#6471] [General]
This discussion is intended to report testing cases for uploading sketches using CDC and DFU using S2, S3 and C3.
The target is to find potential issues and also create a base for a detailed documentation.

**Comment by Sys64736** [comment]
For me, uploading to a Lolin C3 mini with no uart chip in 2.0.2 was ok but no serial output, except a workaround using USBSerial. I also had to open/close and re-open the monitor to get any output after rebooting out of DFU.

In RC-1, it's a big improvement.  Once in DFU mode, uploading works and when the monitor opens, it's in serial output mode already.  I usually only have to take it out of DFU once and it does it automatically everytime after upload. Also, I can now use the Serial command instead of USBSerial using:

  -DARDUINO_USB_MODE=1
  -DARDUINO_USB_CDC_ON_BOOT=1  

Nice work.
